notifications_api_common/viewsets.py

```diff
@@ -162,46 +169,39 @@ def notify(
         # build the content of the notification
         message = self.construct_message(data, instance=instance)
 
+        self.send_notification.delay(status_code, message)
+
+    @staticmethod
+    @celery.current_app.task(bind=True)
+    def send_notification(
+        task,
+        status_code: int,
+        message: dict,
+    ) -> None:
         # build the client from the singleton config. This will raise an
         # exception if the config is not complete. We want this to hard-fail!
         client = NotificationsConfig.get_client()
         if client is None:
             raise RuntimeError("Could not build a client for Notifications API")
 
-        # We've performed all the work that can raise uncaught exceptions that we can
-        # still put inside an atomic transaction block. Next, we schedule the actual
-        # sending block, which allows failures that are logged. Any unexpected errors
-        # here will still cause the transaction to be comitted (in the default
-        # behaviour), but the exception will be visible in the error monitoring (such
-        # as Sentry).
-        #
-        # The _send function is passed down to the scheduler, which is by default to
-        # execute it on transaction commit.
```

Comment:
The core of this comment is still very relevant and I would like to not see this deleted.